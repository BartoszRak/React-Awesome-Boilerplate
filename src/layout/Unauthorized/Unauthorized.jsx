import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'

import UnauthorizedPages from '~pages/UnauthorizedPages'

import styles from './Unauthorized.style'

export function Unauthorized({ classes }) {
  return (
    <div className={classes.root}>
      Unauthorized
      <UnauthorizedPages />
    </div>
  )
}

Unauthorized.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(styles)(Unauthorized)
