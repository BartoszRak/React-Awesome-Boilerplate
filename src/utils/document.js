/* eslint-disable-next-line */
export default document
