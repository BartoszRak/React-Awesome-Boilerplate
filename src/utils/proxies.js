export const makeGetProxy = (obj, missingKeyHandler) => new Proxy(obj, {
  get(target, name) {
    return name in target ? target[name] : missingKeyHandler(target, name)
  },
})

export default makeGetProxy
