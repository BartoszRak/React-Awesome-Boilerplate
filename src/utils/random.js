const rand = (lowest, highest) => {
  const adjustedHigh = highest - lowest + 1
  return Math.floor(Math.random() * adjustedHigh) + parseFloat(lowest)
}
const randChain = (lowest, highest, howLong) => {
  const chain = []
  for (let i = 0; i < howLong; i += 1) {
    chain.push(rand(lowest, highest))
  }
  return chain.join('')
}

export const random = {
  rand,
  randChain,
}

export default random
