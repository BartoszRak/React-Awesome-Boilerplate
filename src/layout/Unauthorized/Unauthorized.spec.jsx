import React from 'react'
import { shallow } from 'enzyme'
import { mocks } from '../../utils/tests'

import { Unauthorized } from './Unauthorized'

const initComponent = overrides => {
  const mockProps = {
    classes: mocks.classesProxy,
  }
  const mockMethods = {}
  const wrapper = shallow(<Unauthorized {...mockProps} {...mockMethods} {...overrides} />)
  return { mockProps, wrapper }
}

describe('Unauthorized: ', () => {
  it('renders without crashing', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toBeTruthy()
  })
  it('should render as expected', () => {
    const { wrapper } = initComponent()
    expect(wrapper).toMatchSnapshot()
  })
})
