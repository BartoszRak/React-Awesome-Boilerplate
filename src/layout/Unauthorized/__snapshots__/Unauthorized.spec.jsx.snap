// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthorized:  should render as expected 1`] = `
<div
  className="root"
>
  Unauthorized
  <UnauthorizedPages />
</div>
`;
